src/OrbitGl/ScopeTree.h

```diff
@@ -109,6 +113,35 @@ ScopeNode<ScopeT>* ScopeTree<ScopeT>::CreateNode(ScopeT* scope) {
   return &new_node;
 }
 
+template <typename ScopeT>
+const ScopeNode<ScopeT>* ScopeTree<ScopeT>::FindScopeNode(const ScopeT& scope) const {
+  for (ScopeNode<ScopeT>* node = root_; node != nullptr;
+       node = node->GetLastChildBeforeOrAtTime(scope.start())) {
+    if (node->Start() == scope.start() && node->End() == scope.end()) {
+      return node;
+    }
+  }
+  return nullptr;
+}
+
+template <typename ScopeT>
+const ScopeT* ScopeTree<ScopeT>::FindNextScopeAtDepth(const ScopeT& scope) const {
+  const ScopeNode<ScopeT>* node = FindScopeNode(scope);
+  auto nodes_at_depth = GetOrderedNodesByDepth().at(node->Depth());
```

Comment:
node could be nullptr, if we expect the function to always be called with a valid scope as argument, then we should `CHECK(node!=nullptr);`, otherwise return nullptr.

Reply:
done